Add integration test for Vector GPU

ThomasRaoux opened the pull request.

ThomasRaoux commented:
Only the second commit is the test first commit is from https://github.com/google/iree-llvm-sandbox/pull/202

ThomasRaoux commented:
Is there an easy way to disable the gpu integration test when building without gpu support?

matthias-springer commented on configure.py:
```diff
@@ -67,6 +67,13 @@ def parse_arguments():
       action=argparse.BooleanOptionalAction,
   )
+  parser.add_argument(
+      "--cuda-runner",
```
What happens when I run `configure.py` without this argument and then run all tests? (run_tests.py)

ThomasRaoux replied:
Good point, I talked to Nicolas about that and I do need to change the script to only enable cuda tests on demand

ThomasRaoux commented:
@nicolasvasilache @matthias-springer I rebased and added a command line option to enable gpu integration tests since they require a gpu and cuda setup. Let me know if this looks okay to you.
